walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,111 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<TimeRange> availableTimeRanges = new ArrayList<TimeRange> ();
+
+    Collection<String> requestAttendees = request.getAttendees();
+    Collection<String> requestOptionalAttendees = request.getOptionalAttendees();
+    Set<String> requestAttendeesSet = new HashSet<> ();
+    Set<String> requestOptionalAttendeesSet = new HashSet<> ();
+    requestOptionalAttendeesSet.addAll(requestOptionalAttendees);
+    requestAttendeesSet.addAll(requestAttendees);
+    requestAttendeesSet.addAll(requestOptionalAttendees);
+    int requestDuration = (int) request.getDuration();
+    
+    Collection<TimeRange> allEventsTimeRanges = new ArrayList<TimeRange> ();
+    for (Iterator currEvent = events.iterator(); currEvent.hasNext(); ){ 
```

Comment:
meganit: event is probably better than currEvent here. We tend not to use "curr"/prev/next unless it provides a distinction. If you were operating over a next, a current, and a previous value, having those prefixes make sense. But when your loop only operates over one of those, you should omit the prefix.